bootstrap/012_task_log.sql

```diff
@@ -0,0 +1,53 @@
+
+create or replace function internal.task_run_id()
+returns text
+AS
+$$
+     SYSTEM$TASK_RUNTIME_INFO('CURRENT_TASK_GRAPH_RUN_GROUP_ID')
+$$;
+
+-- Generic table that tasks should record their execution into.
+CREATE TABLE INTERNAL.TASK_LOG IF NOT EXISTS (task_start timestamp_ltz, success boolean, task_name varchar, object_name varchar,
+    input variant, output variant, task_finish timestamp_ltz, task_run_id text, query_id text);
+
+create or replace procedure internal.migrate_task_logs()
+    returns text
+    language sql
+AS
+BEGIN
+    -- If the old task log tables exist, copy the rows into the new task_log table
+    if (exists (select * from information_schema.tables where table_schema = 'INTERNAL' and table_name = 'TASK_QUERY_HISTORY')) then
+        INSERT INTO INTERNAL.TASK_LOG (task_start, success, task_name, object_name, input, output, task_finish)
+            SELECT tqh.run, tqh.success, 'QUERY_HISTORY_MAINTENANCE', 'QUERY_HISTORY', tqh.input, tqh.output, null
+            FROM INTERNAL.TASK_QUERY_HISTORY tqh;
+        DROP TABLE INTERNAL.TASK_QUERY_HISTORY;
```

Comment:
I don't think we should do this. In this version. I think we should do in future version. If we have a problem and roll back, I think the app would think that a person didn't do full materialization. This would at minimum cause a bunch of extra compute and more likely also cause wrong result (i think). This applies to all the drops/truncates.

Reply:
> If we have a problem and roll back, I think the app would think that a person didn't do full materialization. This would at minimum cause a bunch of extra compute and more likely also cause wrong result (i think).

Yeah, you're exactly right -- it would try to re-materialize things. I'll come up with a different solution that can preserve the old tables.